### ISSUE TITLE
Remember last active windows in project

### ISSUE THREAD

**Issue by krticka** [OpenOrienteering/mapper#1337]
### Steps to reproduce
1. In project dock window with basemaps and with measure tool. 
2. Save and close the project.
3. Reopen the project.

### Actual behaviour
Position of basemaps and measure tool windows is not remembered.

### Expected behaviour
Docked windows are remembered and opened with every project start.

### Configuration
Mapper Version: 0.8.4, dev
Operating System: Win64

**Comment by dg0yt** [comment]
On Linux (Mapper 0.8.4, Qt 5.9.5), all docks *positions* are saved and restored. However, we do not open more than the symbol window with a new map.
Is it really different on Windows, with older *and* with newer Qt ?

**Comment by krticka** [comment]
I have tested it now on dev v20190820.3. If I open project, activate window with basemaps, dock measure measure tool, save the project and open it again, windows which I set up are gone. This is little bit annoying as I use these tools frequently when drawing and have to open them again when I am back to the project.

**Comment by dg0yt** [comment]
What you wrote in the initial post is:

> Actual behaviour
>
>Position of basemaps and measure tool windows is not remembered.

But IIUC you mean:

- *After opening a map again, the dock widgets are not opened again.* 
- When you open them manually, the *do* appear in their last position. 
However, the state is stored in the global settings, not in the map file. This implies that the last position is meaning the last closed Mapper window, not the window with this particular map.

I probably would not add more transient state to the map file because of issue GH-1290. If we want both, we will have to write an extra metadata file side by side with the map file.

**Comment by krticka** [comment]
You are right, I was not precise when defining my initial post. Correct is: After opening a map again, the dock widgets are not opened again. When I open them manually, the do appear in their last position.
Maybe extra metadata file is not needed. Could this be part of undo/redo information? I am personally not cleaning history of changes every drawing session and storing such information there would be enough.
